### PR TITLE
Use the python generator for tests as well

### PR DESCRIPTION
**Pull request by mjcarroll**
Follow up to #233.

This also makes the tests take advantage of the generator.

### CODE REVIEW

File: ros_ign_bridge/bin/ros_ign_bridge_generate_tests [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
```
Comment (ahcorde) [inline ros_ign_bridge/bin/ros_ign_bridge_generate_tests:1]: LICENSE

File: ros_ign_bridge/bin/ros_ign_bridge_generate_tests [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        required=True,
+        help='The basepath of the generated C++ files')
+    parser.add_argument(
+        '--template-dir',
+        required=True,
+        help='The location of the template files')
+    parser.add_argument(
+        '--gz-msgs-ver',
+        required=False,
+        default="0.0.0",
```
Comment (ahcorde) [inline ros_ign_bridge/bin/ros_ign_bridge_generate_tests:43]: can we set this value from the CMAke ? or is it optional ?

File: ros_ign_bridge/test/resource/ign_publisher.cpp.em [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_publisher.cpp.em]: is the year right ?

File: ros_ign_bridge/test/resource/ign_subscriber.cpp.em [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_subscriber.cpp.em]: is the year right ?

File: ros_ign_bridge/test/resource/ign_subscriber.cpp.em [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  /// \param[in] _topic Topic to subscribe.
+
+public:
+  explicit MyTestClass(const std::string & _topic)
+  {
+    EXPECT_TRUE(this->node.Subscribe(_topic, &MyTestClass::Cb, this));
+  }
+
+  /// \brief Member function called each time a topic update is received.
+
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_subscriber.cpp.em]:
```suggestion
  /// \brief Member function called each time a topic update is received.
  /// \param[in] _msg ...
```

File: ros_ign_bridge/test/resource/ign_publisher.cpp.em [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+#include "utils/ign_test_msg.hpp"
+
+/// \brief Flag used to break the publisher loop and terminate the program.
+static std::atomic<bool> g_terminatePub(false);
+
+//////////////////////////////////////////////////
+/// \brief Function callback executed when a SIGINT or SIGTERM signals are
+/// captured. This is used to break the infinite loop that publishes messages
+/// and exit the program smoothly.
+void signal_handler(int _signal)
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_publisher.cpp.em]:
```suggestion
//////////////////////////////////////////////////
/// \brief Function callback executed when a SIGINT or SIGTERM signals are
/// captured. This is used to break the infinite loop that publishes messages
/// and exit the program smoothly.
/// \param[in] _signal ...
void signal_handler(int _signal)
```

File: ros_ign_bridge/test/resource/ign_subscriber.cpp.em [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+#include "utils/ign_test_msg.hpp"
+
+//////////////////////////////////////////////////
+/// \brief A class for testing Ignition Transport topic subscription.
+template<typename IGN_T>
+class MyTestClass
+{
+  /// \brief Class constructor.
+  /// \param[in] _topic Topic to subscribe.
+
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_subscriber.cpp.em]:
```suggestion
```

File: ros_ign_bridge/test/resource/ign_subscriber.cpp.em (hunk 5 above)
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_subscriber.cpp.em]:
```suggestion
```

File: ros_ign_bridge/test/resource/ign_subscriber.cpp.em [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+#include "utils/test_utils.hpp"
+#include "utils/ign_test_msg.hpp"
+
+//////////////////////////////////////////////////
+/// \brief A class for testing Ignition Transport topic subscription.
+template<typename IGN_T>
+class MyTestClass
+{
+  /// \brief Class constructor.
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ign_subscriber.cpp.em]: can you format the file ?

/// Doc
public/private: method name

File: ros_ign_bridge/test/resource/ros_publisher.cpp.em [hunk 9]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ros_publisher.cpp.em]: is the year right ?

File: ros_ign_bridge/test/resource/ros_publisher.cpp.em [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  auto node = rclcpp::Node::make_shared("ros_string_publisher");
+  rclcpp::Rate loop_rate(1);
+
+@[for m in mappings]@
+  // @(m.ros2_string()).
+  auto @(m.unique())_pub = 
+    node->create_publisher<@(m.ros2_type())>("@(m.unique())", 1000);
+  @(m.ros2_type()) @(m.unique())_msg;
+  ros_ign_bridge::testing::createTestMsg(@(m.unique())_msg);
+
```
Comment (ahcorde) [inline ros_ign_bridge/test/resource/ros_publisher.cpp.em:32]:
```suggestion
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: ros_ign_bridge/bin/ros_ign_bridge_generate_tests (hunk 2 above)
Comment (mjcarroll) [reply ros_ign_bridge/bin/ros_ign_bridge_generate_tests:43]: It's being passed in as an argument from CMake: https://github.com/ignitionrobotics/ros_ign/pull/234/files#diff-1c1942d8b5067a8d162d921c4b5bdd95f38e460faf6d9bfdf2e38946c34f1abfR214

This way we don't have to do any sort of generation.

File: ros_ign_bridge/test/resource/ign_subscriber.cpp.em (hunk 8 above)
Comment (mjcarroll) [reply ros_ign_bridge/test/resource/ign_subscriber.cpp.em]: I didn't notice how weird the formatting was when I copied it over.  Fixed it.

File: ros_ign_bridge/test/resource/ros_publisher.cpp.em (hunk 10 above)
Comment (mjcarroll) [reply ros_ign_bridge/test/resource/ros_publisher.cpp.em:32]: This one is intentional so that the generated file is a little easier to read.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**